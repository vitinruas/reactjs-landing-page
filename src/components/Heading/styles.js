import styled, { css } from 'styled-components'

export const Title = styled.h1`
  ${({ theme, appTheme }) =>
    css`
      color: ${appTheme === 'DARK' ? theme.colors.secondaryColor : theme.colors.primaryColor};
    `}
`
