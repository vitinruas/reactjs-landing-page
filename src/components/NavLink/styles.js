import styled, { css } from 'styled-components'

export const Container = styled.nav`
  ${({ theme, appTheme }) => css`
    background: ${appTheme === 'DARK' ? theme.colors.secondaryColor : theme.colors.primaryColor};
    display: flex;
    flex-flow: wrap row;

    @media ${theme.medias.lteMedium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`
