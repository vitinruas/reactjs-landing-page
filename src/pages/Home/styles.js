import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${(props) => {
    return css`
      background: ${props.background};
    `
  }}
`
