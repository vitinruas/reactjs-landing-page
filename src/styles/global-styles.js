import { createGlobalStyle, css } from 'styled-components'

const GlobalStyles = createGlobalStyle`
  * {
    padding: 0;
    margin: 0;
    box-sizing: border-box;
  }


  @media (min-width: 320px) and (max-width: 480px) {
    html {
      font-size: 75%; // 12
    }
  }

  @media (min-width: 481px) and (max-width: 768px) {
    html {
      font-size: 81.25%; // 13
    }
  }

  @media (min-width: 769px) and (max-width: 1024px) {
    html {
      font-size: 87.5%; // 14
    }
  }

  @media (min-width: 1025px) and (max-width: 1200px) {
    html {
      font-size: 93.75%; // 15
    }
  }

  @media (min-width: 1201) {
    html {
      font-size: 16px; // 16
    }
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) =>
      css`
        ${theme.fonts.family.text}
      `}

  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) =>
      css`
        ${theme.fonts.family.title}
      `}
  }
`

export { GlobalStyles }
