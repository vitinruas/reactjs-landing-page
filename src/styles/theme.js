export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#FFFFFF',
    tertiaryColor: '#dc143c',

    lightBg: '#FFF',
    darkBg: '#0A1128',
  },
  fonts: {
    family: {
      text: "'Open Sans', sans-serif",
      title: "'Montserrat', sans-serif",
    },
    sizes: {
      small: '1.6rem',
      medium: '2.0rem',
      xmedium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      big: '4.8rem',
      xbig: '5.6rem',
      huge: '6.4rem',
      xhuge: '7.2rem',
    },
  },
  spacings: {
    small: '0.5rem',
    medium: '1rem',
    xmedium: '1.5rem',
    large: '2rem',
    xlarge: '2.5rem',
    big: '3.5rem',
    xbig: '4rem',
    huge: '4.5rem',
    xhuge: '5rem',
  },

  medias: {
    lteMedium: '(max-width: 768px)',
  },
}
