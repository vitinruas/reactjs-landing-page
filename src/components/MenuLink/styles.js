import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'

export const Container = styled.a`
  ${({ theme, appTheme }) => css`
    display: block;
    font-size: ${theme.fonts.sizes.small};
    padding: ${theme.spacings.medium};
    color: ${appTheme === 'DARK' ? theme.colors.primaryColor : theme.colors.secondaryColor};
    position: relative;
    text-decoration: none;

    &:hover {
      cursor: pointer;
    }

    &::after {
      content: '';
      position: absolute;
      left: 50%;
      bottom: 8px;
      transform: translateX(-50%);
      width: 0;
      height: 0.2rem;
      background-color: ${theme.colors.tertiaryColor};
      transition: all 0.5s;
    }
    &:hover::after {
      width: 50%;
    }
  `}
`
