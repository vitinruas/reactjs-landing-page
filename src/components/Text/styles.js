import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'
export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.fonts.sizes.medium};
  `}
`
