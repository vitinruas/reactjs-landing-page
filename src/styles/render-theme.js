import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { GlobalStyles } from './global-styles'
import { theme } from './theme'

export const renderTheme = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>,
  )
}
