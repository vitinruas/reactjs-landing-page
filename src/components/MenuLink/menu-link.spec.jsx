import { screen } from '@testing-library/react'
import { MenuLink } from '.'
import { renderTheme } from '../../styles/render-theme'

describe('<MenuLink />', () => {
  test('should only render <MenuLink /> with a text and link', () => {
    renderTheme(<MenuLink link="any_link">any_text</MenuLink>)
    const link = screen.queryByRole('link', { name: /any_text/i })
    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('href', 'any_link')
  })
})
