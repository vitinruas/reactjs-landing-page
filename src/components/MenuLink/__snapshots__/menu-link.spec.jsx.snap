// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should take a snapshot 1`] = `
.c0 {
  display: block;
  font-size: 1.6rem;
  padding: 1rem;
  color: #FFFFFF;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  cursor: pointer;
}

.c0::after {
  content: '';
  position: absolute;
  left: 50%;
  bottom: 8px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c0:hover::after {
  width: 50%;
}

<a
  class="c0"
  href="any_link"
  target="_self"
>
  any_text
</a>
`;
